Content.Shared/SubFloor/SubFloorHideSystem.cs

```diff
@@ -152,10 +168,28 @@ private void UpdateEntity(EntityUid uid)
             }
 
             // Update normally.
-            UpdateEntity(uid, IsSubFloor(grid, grid.TileIndicesFor(transform.Coordinates)));
+            bool isSubFloor = IsSubFloor(grid, grid.TileIndicesFor(transform.Coordinates));
+            UpdateEntity(uid, isSubFloor);
+        }
+
+        // Toggles an enumerable set of entities to display.
+        public void ToggleSubfloorEntities(IEnumerable<EntityUid> entities, bool visible, EntityUid? uid = null, IEnumerable<object>? appearanceKeys = null)
+        {
+            foreach (var entity in entities)
+            {
+                if (!EntityManager.HasComponent<SubFloorHideComponent>(entity))
+                    continue;
+
+                var transform = EntityManager.GetComponent<TransformComponent>(entity);
+                if (_mapManager.TryGetGrid(transform.GridID, out var grid))
+                {
+                    bool isSubFloor = IsSubFloor(grid, grid.TileIndicesFor(transform.Coordinates));
+                    UpdateEntity(entity, visible, uid, appearanceKeys);
+                }
+            }
         }
 
-        private void UpdateEntity(EntityUid uid, bool subFloor)
+        private void UpdateEntity(EntityUid uid, bool subFloor, EntityUid? revealedUid = null, IEnumerable<object>? appearanceKeys = null)
         {
             // We raise an event to allow other entity systems to handle this.
             var subFloorHideEvent = new SubFloorHideEvent(subFloor);
```

Comment:
I think this event is completely unused? Maybe it was initially added so t-ray scanners could eventually use it? Either way, maybe just remove it for now.

Reply:
yeah, it was left there by whoever was working on subfloorhidesystem before me - i might just remove it if there's no real reason to have it

Reply:
I can't think of a reason why some other system would prevent a subfloor entity from being hidden (other than t-ray scanners), so I don't think its needed.